build(datasets): Add official support for python 3.13

ravi-kumar-pilla opened the pull request:
## Description

Related to https://github.com/kedro-org/kedro/issues/4267

**NOTE:**

Kedro-datasets - blocked by tensorflow
- Tensorflow - https://github.com/tensorflow/tensorflow/issues/78774
- Transformers sentencepiece - https://github.com/google/sentencepiece/issues/1069
- BioPython - https://github.com/biopython/biopython/issues/4883
- Geopandas - https://github.com/geopandas/geopandas/issues/3400


## Development notes

- Update CI build to include python 3.13

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes
- [ ] Received approvals from at least half of the TSC (required for adding a new, non-experimental dataset)

merelcht commented on kedro-datasets/pyproject.toml:
```diff
@@ -186,7 +186,7 @@ prophet = ["kedro-datasets[prophet]"]
 pytorch = ["kedro-datasets[pytorch-dataset]"]
 
-rioxarray-geotiffdataset = ["rioxarray>=0.15.0"]
+rioxarray-geotiffdataset = ["rioxarraπtey>=0.15.0"]
```
Is this a typo? 😅

noklam commented:
Took me a while to notice the blocker is `ibis` as it is updated in `main` two months ago but haven't done a release. Pending for response, we can either wait for the next `ibis` release, or drop `ibis` support for Python 3.13 (same as Tensorflow) until they officially support it.

ravi-kumar-pilla commented:
Hi @astrojuanlu and @noklam, 

There seems to be some more libraries which do not yet support python 3.13. I have updated the list [here](https://github.com/kedro-org/kedro-plugins/pull/938#issue-2680546897). I have commented the pytests for few and also added conditional package installations in pyproject.toml. I feel this is becoming a bit messy and would like to get your thoughts. 

Can we do a kedro-datasets 6.0.0 release without python 3.13 support ? If not do you see any other way to implement this ?

Thank you

astrojuanlu commented:
> There seems to be some more libraries which do not yet support python 3.13. I have updated the list https://github.com/kedro-org/kedro-plugins/pull/938#issue-2680546897. I have commented the pytests for few and also added conditional package installations in pyproject.toml. I feel this is becoming a bit messy and would like to get your thoughts.

I agree with making the 6.0.0 release without Python 3.13 support 👍🏼

merelcht commented:
I suggest closing this PR for now and create a new one when all blockers are resolved.

ravi-kumar-pilla commented:
Closing this PR for now as we have unresolved dependencies
